pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -361,7 +361,7 @@ func ParseResources(cecNamespace string, cecName string, anySlice []cilium_v2.XD
 	// Do this only after all other possible error cases.
 	for _, listener := range resources.Listeners {
 		if listener.GetAddress() == nil {
-			port, err := portAllocator.AllocateProxyPort(listener.Name, false)
+			port, err := portAllocator.AllocateProxyPort(listener.Name, false, true)
```

Comment:
It kind of is already, you can specify an address and port in the Listener spec. This path is only executed when there is no address in the Listener spec, meaning that "let each Cilium Agent pick a local port". With this change that now includes the fact the locally allocated port is listened only on localhost interfaces. The locally allocated port is essentially random, so it is different on each node and is not exposed in any user facing API. Given this it would be hard to send traffic to that port from the outside of the node anyway, so listening on localhost only does not change much. Other than have one less open port on external interfaces potentially vulnerable for port scanning.